ion-schema/src/system.rs

```diff
@@ -1064,6 +1064,11 @@ impl Resolver {
 }
 
 /// Provides functions for instantiating instances of [`Schema`].
+///
+/// [`SchemaSystem`] is *[Send and Sync]* i.e. it is safe to send it to another thread and to be shared between threads.
+/// For cases when one does need thread-safe interior mutability, they can use the explicit locking via [`std::sync::Mutex`] and [`std::sync::RwLock`].
```

Comment:
For what sort of cases would one need thread-safe interior mutability for `SchemaSystem`?

Reply:
I was thinking of modifying the list `DocumentAuthority`s that are part of `SchemaSystem` and for `DocumentAuthority`s modifying the authorities content specially for `MapDocumentAuthority` which can store a schema content.